package edu.princeton.jetstream;

import "jetstream_types.proto";

message AlterTopo {
  repeated TaskMeta toStart = 1;
  repeated TableMeta toCreate = 2;
  repeated TaskID taskToStop = 3; 
  repeated string tableToStop = 4; 
  repeated Edge edges = 5;
}


/* Name isn't sufficient descriptive */
message ServerRequest {
  enum Type {GET_NODES = 1; DEPLOY = 3 ; COUNT_NODES = 4;}
  required Type type = 1;

  optional AlterTopo alter = 3;
}


/* Name isn't sufficient descriptive */
message ServerResponse {
  optional string error = 147; //use unusual number to catch errors on parsing
  repeated NodeID nodes = 148; //in response to a get_nodes or get_node command

  optional int32 count_nodes = 149;
}